internal/cmd/egctl/translate_test.go

```diff
@@ -26,6 +98,7 @@ func TestTranslate(t *testing.T) {
 		resourceType string
 		extraArgs    []string
 		expect       bool
+		filterFunc   func(string) string
```

Comment:
these filter make it very hard for me to understand what the final output will look like

Reply:
I would like to remove it, but unfortunately, there are some timestamp fields like `lastTransitionTime: "2023-03-21T12:03:11Z"` which need to be stripped from the expected data.

Reply:
we skip timestamp in many places, here's an example
https://github.com/envoyproxy/gateway/blob/384577334dea2c0ba28800643a2b94fea4eed244/internal/gatewayapi/translator_test.go#L103

Reply:
thanks for the clarification, I understand the issue better now, do you prefer this approach instead of creating some typed field instead of `[]interface{}{}` ?

Reply:
Yes, I will remove `[]interface{}{}` from the filterFunc.